File: frontend/src/types/admin/admin-report-detail-dto.ts
import { ReportResolution } from "../report/report-resolution";
import { ReportTargetAction } from "../report/report-target-action";
import { AdminReportDTO, ProfileSummaryDTO } from "./admin-report-dto";

export interface ReportTargetSnapshotDTO {
  targetTitle?: string | null;
  targetPreview?: string | null;
  targetOwnerDisplayName?: string | null;
  targetOwnerAvatarUrl?: string | null;
  route?: string | null;
  createdAt?: string | null;
  updatedAt?: string | null;
  isDeletedAtReportTime: boolean;
}

export interface AdminReportDetailDTO {
  report: AdminReportDTO;
  reportedVersion?: unknown;
  currentTarget?: unknown;
  currentTargetState?: {
    unavailable: boolean;
    deleted: boolean;
    deletedAt?: string | null;
    hidden: boolean;
    moderationStatus?: string | null;
    private: boolean;
    suspended: boolean;
    suspendedUntil?: string | null;
    parentUnavailable: boolean;
    parentDeleted: boolean;
    parentHidden: boolean;
    parentModerationStatus?: string | null;
  } | null;
  reporter?: ProfileSummaryDTO | null;
  targetOwner?: ProfileSummaryDTO | null;
  targetSnapshotJson?: string | null;
  targetSnapshot?: ReportTargetSnapshotDTO | null;
  moderatorNote?: string | null;
  resolution?: ReportResolution | null;
  resolutionNote?: string | null;
  resolvedById?: string | null;
  resolvedBy?: ProfileSummaryDTO | null;
  resolvedAt?: string | null;
  targetAction?: ReportTargetAction | null;
}

File: platform-core-service/platform-core-service.Common/Models/DTOs/EntityDTO/Report/AdminReportDetailDTO.cs
using platform_core_service.Common.Utils.Enums;

namespace platform_core_service.Common.Models.DTOs.EntityDTO.Report
{
    public class AdminReportDetailDTO
    {
        public AdminReportDTO Report { get; set; } = null!;
        public object? ReportedVersion { get; set; }
        public object? CurrentTarget { get; set; }
        public ProfileSummaryDTO? Reporter { get; set; }
        public ProfileSummaryDTO? TargetOwner { get; set; }
        public string? TargetSnapshotJson { get; set; }
        public ReportTargetSnapshotDTO? TargetSnapshot { get; set; }
        public AdminReportTargetStateDTO? CurrentTargetState { get; set; }
        public string? ModeratorNote { get; set; }
        public ReportResolution? Resolution { get; set; }
        public string? ResolutionNote { get; set; }
        public string? ResolvedById { get; set; }
        public ProfileSummaryDTO? ResolvedBy { get; set; }
        public DateTimeOffset? ResolvedAt { get; set; }
        public ReportTargetAction? TargetAction { get; set; }
    }

    public class AdminReportTargetStateDTO
    {
        public bool Unavailable { get; set; }
        public bool Deleted { get; set; }
        public DateTimeOffset? DeletedAt { get; set; }
        public bool Hidden { get; set; }
        public string? ModerationStatus { get; set; }
        public bool Private { get; set; }
        public bool Suspended { get; set; }
        public DateTimeOffset? SuspendedUntil { get; set; }
        public bool ParentUnavailable { get; set; }
        public bool ParentDeleted { get; set; }
        public bool ParentHidden { get; set; }
        public string? ParentModerationStatus { get; set; }
    }
}
